Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemos/GazeContingent/FixWindowBufferedSamples/EyeLink_FixWindowBufferedSamples.m

```diff
@@ -0,0 +1,538 @@
+        Eyelink('Command', 'link_sample_data  = LEFT,RIGHT,GAZE,GAZERES,AREA,HTARGET,STATUS,INPUT');
+    else
+        Eyelink('Command', 'file_sample_data  = LEFT,RIGHT,GAZE,HREF,RAW,AREA,GAZERES,BUTTON,STATUS,INPUT');
+        Eyelink('Command', 'link_sample_data  = LEFT,RIGHT,GAZE,GAZERES,AREA,STATUS,INPUT');
+    end
+    
+    
+    %% STEP 3: OPEN GRAPHICS WINDOW
+    
+    % Open experiment graphics on the specified screen
+    if isempty(screenNumber)
+        screenNumber = max(Screen('Screens')); % Use default screen if none specified
+    end
+    [window, wRect] = Screen('OpenWindow', screenNumber, [128 128 128]); % Open graphics window
+    Screen('Flip', window);
+    % Return width and height of the graphics window/screen in pixels
+    [width, height] = Screen('WindowSize', window);
+    
+    
+    %% STEP 4: SET CALIBRATION SCREEN COLOURS; PROVIDE WINDOW SIZE TO EYELINK HOST & DATAVIEWER; SET CALIBRATION PARAMETERS; CALIBRATE
+    
+    % Provide EyeLink with some defaults, which are returned in the structure "el".
+    el = EyelinkInitDefaults(window);
+    % set calibration/validation/drift-check(or drift-correct) size as well as background and target colors. 
+    % It is important that this background colour is similar to that of the stimuli to prevent large luminance-based 
+    % pupil size changes (which can cause a drift in the eye movement data)
+    el.calibrationtargetsize = 3;% Outer target size as percentage of the screen
+    el.calibrationtargetwidth = 0.7;% Inner target size as percentage of the screen
+    el.backgroundcolour = [128 128 128];% RGB grey
+    el.calibrationtargetcolour = [0 0 0];% RGB black
+    % set "Camera Setup" instructions text colour so it is different from background colour
+    el.msgfontcolour = [0 0 0];% RGB black
+    % You must call this function to apply the changes made to the el structure above
+    EyelinkUpdateDefaults(el);
+    
+    % Set display coordinates for EyeLink data by entering left, top, right and bottom coordinates in screen pixels
+    Eyelink('Command','screen_pixel_coords = %ld %ld %ld %ld', 0, 0, width-1, height-1);
+    % Write DISPLAY_COORDS message to EDF file: sets display coordinates in DataViewer
+    % See DataViewer manual section: Protocol for EyeLink Data to Viewer Integration > Pre-trial Message Commands
+    Eyelink('Message', 'DISPLAY_COORDS %ld %ld %ld %ld', 0, 0, width-1, height-1);    
+    % Set number of calibration/validation dots and spread: horizontal-only(H) or horizontal-vertical(HV) as H3, HV3, HV5, HV9 or HV13
+    Eyelink('Command', 'calibration_type = HV9'); % horizontal-vertical 9-points
+    % Allow a supported EyeLink Host PC button box to accept calibration or drift-check/correction targets via button 5
+    Eyelink('Command', 'button_function 5 "accept_target_fixation"');
+    % Hide mouse cursor
+    HideCursor(screenNumber);

```

Comment:
General best approach is to pass in the onscreen window handle, although this also typically works.

Reply:
resolved in 0abf0cbe36691edea3a6d49a87b79149ad7e6f94